### PR TITLE
Allow users to configure passports to support claimable funds

### PR DESCRIPTION
**Pull request by dvargasfuertes**
https://app.clarity.so/cabin/view/3039c279-2ee2-4da2-a604-dc1c23d5010c

### CODE REVIEW

File: packages/app/pages/index.tsx [hunk 1]
```diff
@@ -90,12 +98,19 @@ interface IMembershipProps {
   supply: number;
   price: string;
   metadataHash: string;
+  claimable: boolean;
```
Comment (dvargasfuertes) [inline packages/app/pages/index.tsx:101]: Not a big fan of this name

File: packages/app/pages/index.tsx [hunk 2]
```diff
@@ -210,19 +241,23 @@ const MembershipCard = (props: IMembershipCardProps) => {
...
-                id="acolor" 
-                name="acolor" 
+              <input
+                type="color"
+                id="acolor"
+                name="acolor"
                 value={accColor || "#324841"}
-                onChange={(e) => setAccColor(e.target.value)}></input>
+                onChange={(e) => setAccColor(e.target.value)}
+              ></input>
```
Comment (dvargasfuertes) [inline packages/app/pages/index.tsx:261]: sorry about this noise, I have an autoformatter in my code editor that formats on save

File: packages/contracts/contracts/Passport.sol [hunk 3]
```diff
@@ -15,19 +15,21 @@ contract Passport is ERC721, Ownable {
   uint256 public supply;
   uint256[] public tokenIds;
   string public metadataHash;
+  bool public claimable;
   event Purchase(address owner, uint256 price, uint256 id, string uri);
-  constructor(address owner_, string memory name_, string memory symbol_, uint256 _supply, uint256 _price, string memory _metadataHash) ERC721(name_, symbol_) {
+  constructor(address owner_, string memory name_, string memory symbol_, uint256 _supply, uint256 _price, string memory _metadataHash, bool _claimable) ERC721(name_, symbol_) {
```
Comment (dvargasfuertes) [inline packages/contracts/contracts/Passport.sol]: I'm going to wait for that Royalty PR to merge and will rebase this PR on it

File: packages/contracts/contracts/PassportFactory.sol [hunk 4]
```diff
@@ -18,9 +18,9 @@ contract PassportFactory is Ownable {
     address passport
   );
 
-  function create(string memory name_, string memory symbol_, uint256 supply, uint256 price, string memory metadataHash) public {
+  function create(string memory name_, string memory symbol_, uint256 supply, uint256 price, string memory metadataHash, bool claimable) public {
     require(supply > 0, "Required to mint at least 1 passport");
-    Passport passport = new Passport(msg.sender, name_, symbol_, supply, price, metadataHash);
+    Passport passport = new Passport(msg.sender, name_, symbol_, supply, price, metadataHash, claimable);
```
Comment (dvargasfuertes) [inline packages/contracts/contracts/PassportFactory.sol]: I wonder what's the max number of parameters a constructor/method could handle in Solidity lol

**Review by dvargasfuertes — COMMENTED** [review]
ccing @ssaitejar96 for review

File: packages/contracts/contracts/PassportFactory.sol (hunk 4 above)
Comment (ssaitejar96) [reply packages/contracts/contracts/PassportFactory.sol]: We have more parameters coming in lol - royalties and private

File: packages/contracts/contracts/Passport.sol (hunk 3 above)
Comment (ssaitejar96) [reply packages/contracts/contracts/Passport.sol]: Updated the royalties PR.

File: packages/contracts/contracts/Passport.sol [hunk 5]
```diff
@@ -51,8 +55,13 @@ contract Passport is ERC721, Ownable {
     emit Purchase(msg.sender, price, _id, tokenURI(_id));
   }
 
-  function get() public view returns(string memory, string memory, uint256, uint256, string memory) {
-    return (name(), symbol(), supply, price, metadataHash);
+  function get() public view returns(string memory, string memory, uint256, uint256, string memory, bool) {
+    return (name(), symbol(), supply, price, metadataHash, claimable);
```
Comment (ssaitejar96) [inline packages/contracts/contracts/Passport.sol]: When you are resolving conflicts with royalties PR on this line, please make sure to put claimable before royaltyPcnt as I see that you've used p[5] in the UI side code.

File: packages/app/pages/index.tsx [hunk 6]
```diff
@@ -376,7 +439,7 @@ const CreateMembershipModal = ({
       const weiPrice = web3.utils.toWei(price, "ether");
       return new Promise<void>((resolve, reject) =>
         contractInstance.methods
-          .create(name, symbol, quantity, weiPrice, metadataHash)
+          .create(name, symbol, quantity, weiPrice, metadataHash, claimable)
```
Comment (ssaitejar96) [inline packages/app/pages/index.tsx]: You can pass royaltyPcnt as 0 for now when you rebase with royalties PR.  I will be pushing a PR for UI side royalty changes once this PR is merged.

**Review by ssaitejar96 — APPROVED** [review]
Overall looks great! Just left a couple of comments for when you rebase with royalties PR.

File: packages/contracts/contracts/Passport.sol (hunk 5 above)
Comment (dvargasfuertes) [reply packages/contracts/contracts/Passport.sol]: I'd rather keep the order in the `get` and `constructor` consistent. Could easily update the UI to use `p[6]`

**Comment by dvargasfuertes** [comment]
Gonna retap you for review to look over royaltyInfo changes in `Passport.sol`

File: packages/contracts/contracts/Passport.sol [hunk 7]
```diff
@@ -55,32 +61,25 @@ contract Passport is ERC721, Ownable {
...
-      royaltyAmount = (value * royalties.amount) / 10000;
-  }
-
-  /// @dev Sets token royalties
-  /// @param recipient recipient of the royalties
-  /// @param value percentage (using 2 decimals - 10000 = 100, 0 = 0)
-  function _setRoyalties(address recipient, uint256 value) internal {
-      require(value <= 10000, 'ERC2981Royalties: Too high');
-      _royalties = RoyaltyInfo(recipient, uint24(value));
+      receiver = _owner;
```
Comment (ssaitejar96) [inline packages/contracts/contracts/Passport.sol]: This should be receiver = royaltyRecipient, right?

**Review by ssaitejar96 — CHANGES_REQUESTED** [review]
One blocking comment.

File: packages/contracts/contracts/Passport.sol (hunk 7 above)
Comment (dvargasfuertes) [reply packages/contracts/contracts/Passport.sol]: see, that's why I rerequested review

**Review by ssaitejar96 — APPROVED** [review]

**Pull request merged**